### PR TITLE
Add DOM Merge Conflict workshop

### PR DESCRIPTION
**Pull request by girdenis-p**
# DOM Merge Conflicts Workshop

## Overview

This workshop is intended to be used during the Launch module. The workshop is intended to give trainees more experience with how merge conflicts arise and how to resolve them. The premise of the tasks is that, when implemented by two separate trainees, they have a high chance of creating a merge conflict.

## Previous Trial Results

This workshop was trialled by @leilafaez, @levilu800b, and @Sandra-Duarte. And I invite them to help review this pull request.

As per feedback, I have:
- Updated the README and instructions to be more clear.
- Adjusted the *conditional-rendering* task so that it is more likely that both trainees change the same line.
  - This was achieved by changing the task to ask trainees to modify the arguments of the ProfilePreview function.
- Added a new task *processing-page*.
  - This is intended to create a merge conflict when the trainees both modify logic of how the promises are processed.
  
## Side Note

This workshop may benefit from being created as a stand-alone repo similar to how previous merge conflict workshops have been. This would be purely for ease of use. I would be happy to recreate this pull request on an empty new repo if you agree to making a stand-alone repo.

### CODE REVIEW

**Review by leilafaez — APPROVED** [review]

**Comment by leilafaez** [comment]
> Can someone please approve this - if you are happy with it? @leilafaez @levilu800b @Sandra-Duarte

Hi @kfklein15. I approved it but it needs 3 other reviews as well

**Review by Dedekind561 — CHANGES_REQUESTED** [review]
Hi @girdenis-p,
Thanks for all your effort on this workshop - some really interesting activities here! We'd like to get this in ASAP so we can start iterating on this and even use it as early as the databases module. I've suggested a few changes to simplify some of the instructions and make the tests clearer.

File: dom-merge-conflict/tasks/conditional-rendering/test/profilePreview.test.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    beforeEach(() => {
+      profile = {
+        pictureSrc: "/null",
+        name: "N/A",
+        bio: "N/A",
+      };
+    });
+
+    test("only notice that preview is unavailable is displayed with shortform on, available off", () => {
+      profile.available = false;
```
[the remark this answers is not in the document]
Comment (girdenis-p) [reply dom-merge-conflict/tasks/conditional-rendering/test/profilePreview.test.js:57]: Yes, it was originally intended for all properties to be passed through via a single object, but that was changed so this is legacy code that I will remove.

File: dom-merge-conflict/tasks/buttons-and-counter/instructions.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+## Task for Person 1
+
+Create and switch to a new branch named `split-app-components`:
+
+```
+git switch -c split-app-components
+```
+
+Split the header and main content into two seperate files. Use the new functions to refactor the App function.
```
[the remark this answers is not in the document]
Comment (girdenis-p) [reply dom-merge-conflict/tasks/buttons-and-counter/instructions.md:13]: I intended, the trainees to end up with three files in total e.g. header.js, main.js, and app.js. I will update the instructions to reflect this in a similar manner you suggested here.

[the remark this answers is not in the document]
Comment (girdenis-p) [reply dom-merge-conflict/tasks/buttons-and-counter/instructions.md]: Resolved by 59f000a7608969dd26b9fb2f5aa418961c0678cc

File: dom-merge-conflict/tasks/buttons-and-counter/src/app.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  const header = document.createElement("header");
+  header.innerHTML = `
+        <h1>Number Counter</h1>
+        <p>A simple counter. Press increment to increase the count by one.</p>
+    `;
+  body.appendChild(header);
+
+  const buttonAndCounter = document.createElement("main");
+  buttonAndCounter.innerHTML = `
```
[the remark this answers is not in the document]
Comment (girdenis-p) [reply dom-merge-conflict/tasks/buttons-and-counter/src/app.js]: Resolved by 784b8edf8a50653ee46ddf28a2fb0e57ed4a1c75

File: dom-merge-conflict/tasks/conditional-rendering/test/profilePreview.test.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+    test("contains 3 elements", () => {
+      expect(container.childElementCount).toBe(3);
+    });
+  });
+
+  describe.skip("available option", () => {
+    let profile = {};
+
+    beforeEach(() => {
```
[the remark this answers is not in the document]
Comment (girdenis-p) [reply dom-merge-conflict/tasks/conditional-rendering/test/profilePreview.test.js]: I have removed the beforeEach and followed your suggestion. Resolved by 476660b316e99ca77301ddfab087071b6e808fdd.

File: dom-merge-conflict/tasks/conditional-rendering/test/profilePreview.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (girdenis-p) [reply dom-merge-conflict/tasks/conditional-rendering/test/profilePreview.test.js]: Resolved by 476660b316e99ca77301ddfab087071b6e808fdd.

File: dom-merge-conflict/tasks/conditional-rendering/test/profilePreview.test.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+      name: "Jane Doe",
+      bio: "Duis porta neque sed eros.",
+    };
+    Object.freeze(profile);
+
+    const container = ProfilePreview(profile, {
+      shortForm: true,
+    });
+
+    test("contains provided image", () => {
```
[the remark this answers is not in the document]
Comment (girdenis-p) [reply dom-merge-conflict/tasks/conditional-rendering/test/profilePreview.test.js]: I have merged them into one test. Resolved by 476660b316e99ca77301ddfab087071b6e808fdd.

**Review by SallyMcGrath — APPROVED** [review]

**Pull request merged**